derp/derp_server.go

```diff
@@ -164,11 +165,11 @@ type Server struct {
 	// remote). If the value is non-nil, it's remote (+ maybe also
 	// local).
 	clientsMesh map[key.NodePublic]PacketForwarder
-	// sentTo tracks which peers have sent to which other peers,
-	// and at which connection number. This isn't on sclient
-	// because it includes intra-region forwarded packets as the
-	// src.
-	sentTo map[key.NodePublic]map[key.NodePublic]int64 // src => dst => dst's latest sclient.connNum
+	// peerGoneWatchers is the set of watchers that subscribed to a
+	// peer disconnecting from the region overall. When a peer
+	// is gone from the region, we notify all of these watchers,
+	// calling their funcs in a new goroutine.
+	peerGoneWatchers map[key.NodePublic]set.HandleSet[func(key.NodePublic)]
```

Comment:
note that instead of set.HandleSet this could've also been a `set.Set[*sclient]` but I didn't want the Server/watch mechanism having that much access to the `*sclient`. It seemed easier (to me, at least) to reason about it if the watcher supplied its own callback so there was less to understand on the waker side and all the wakee side was together (the call site registering the watcher also saying where the code goes next)